src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -1097,4 +1111,19 @@ public InvocationRequest tryRemoveInvocation(String id) {
             return handlers.get(0).getClasses();
         }
     }
+
+    @Override
+    public void close() throws Exception {
+        stop().subscribe(() -> {
```

Comment:
Shouldn't this block on stop() and cleanup(), so when close() returns the HubConnection is completely closed?

Reply:
Yeah, I guess this is like calling Dispose on an object with async close, you need to block on the task.